pkg/test/istio_io/script_step.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package istio_io
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	testEnv "istio.io/istio/pkg/test/env"
+	"istio.io/istio/pkg/test/scopes"
+	"istio.io/istio/pkg/test/util/file"
+)
+
+var _ step = &scriptStep{}
+
+type scriptStep struct {
+	script   FileSelector
+	verifier Verifier
+}
+
+func newScriptStep(script FileSelector, verifier Verifier) step {
+	return &scriptStep{
+		script:   script,
+		verifier: verifier,
+	}
+}
+
+func (s *scriptStep) Run(ctx Context) {
+	// Copy the script to the output directory.
+	script := s.script.SelectFile(ctx)
+	scriptSourceDir, scriptFileName := path.Split(script)
+	if err := file.Copy(script, path.Join(ctx.OutputDir, scriptFileName)); err != nil {
+		ctx.Fatalf("failed copying input files for step: %v", err)
+	}
+
+	scopes.CI.Infof("Executing %s", s.script)
+
+	content, err := ioutil.ReadFile(script)
+	if err != nil {
+		ctx.Fatal("test framework failed to read script: %s", err)
+	}
+
+	//replace @.*@ with the correct paths
+	atMatch := regexp.MustCompile("@.*@")
+	content = atMatch.ReplaceAllFunc(content, func(input []byte) []byte {
+		trimmed := input[1 : len(input)-1]
+		return []byte(path.Join(testEnv.IstioSrc, string(trimmed)))
+	})
```

Comment:
We'll should call copy on any substituted files here as well.

Reply:
Not sure I understand what you're asking here ... are you referring to the replace function?

Reply:
Yeah, scripts can execute other scripts and apply Yaml files. Currently those don't get coppied

Reply:
Don't think it matters. Was just thinking thoroughness but those won't show up to the user.